allure-rest-assured/src/test/java/io/qameta/allure/restassured/AllureRestAssuredTest.java

```diff
@@ -23,46 +23,65 @@
 import io.qameta.allure.model.TestResult;
 import io.qameta.allure.test.AllureResults;
 import io.restassured.RestAssured;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static io.qameta.allure.test.RunUtils.runWithinTestContext;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class AttachmentArgumentProvider implements ArgumentsProvider {
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+
+        return Stream.of(
+                arguments(ImmutableList.of("Request", "HTTP/1.1 200 OK"), new AllureRestAssured()),
+                arguments(ImmutableList.of("Allure Request", "Allure Response"), new AllureRestAssured().setRequestAttachmentName("Allure Request").setResponseAttachmentName("Allure Response")),
+                arguments(ImmutableList.of("Request", "Allure Response"), new AllureRestAssured().setResponseAttachmentName("Allure Response")),
+                arguments(ImmutableList.of("Allure Request", "HTTP/1.1 200 OK"), new AllureRestAssured().setRequestAttachmentName("Allure Request"))
+        );
+    }
+}
 
 /**
  * @author charlie (Dmitry Baev).
  */
 @SuppressWarnings("unchecked")
 class AllureRestAssuredTest {
 
-    static Stream<String> attachmentNameProvider() {
-        return Stream.of("Request", "HTTP/1.1 200 OK");
-    }
-
     @ParameterizedTest
-    @MethodSource(value = "attachmentNameProvider")
-    void shouldCreateAttachment(final String attachmentName) {
+    @ArgumentsSource(AttachmentArgumentProvider.class)
+    void shouldCreateAttachment(final List<String> attachmentNames, AllureRestAssured filter) {
```

Comment:
Let's follow the same convention for each argument. If you use `final` for one, it should be the same for the other.